### PR TITLE
LTSR Security Patch for June '24

### PR DESCRIPTION
**Pull request by bluzarraga**
updates net, protobuf and apimachinery pkgs
Addresses the following issues
- https://github.ibm.com/IBMPrivateCloud/roadmap/issues/63048
- https://github.ibm.com/IBMPrivateCloud/roadmap/issues/63164

### CODE REVIEW

**Comment by bluzarraga** [comment]
well thats not good

File: controllers/secretshare_controller.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ type SecretShareReconciler struct {
 // +kubebuilder:rbac:groups=,resources=secret;configmap,verbs=get;update;patch;create;delete;list;watch
 
 // Reconcile reads that state of the cluster for a SecretShare object and makes changes based on the state read
-func (r *SecretShareReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	ctx := context.Background()
+func (r *SecretShareReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctx = context.Background()
```
Comment (bluzarraga) [inline controllers/secretshare_controller.go]: I get a warning for overwriting the variable at the beginning here. Should I remove this ctx = context.Background() line? I'm not completely sure how this works so don't want to introduce a regression

Comment (bluzarraga) [reply controllers/secretshare_controller.go]: never mind, this break the linter so will remove

**Comment by bluzarraga** [comment]
Ok this is ready for review now

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**